aea/crypto/cosmos.py

```diff
@@ -397,6 +397,25 @@ def _try_get_balance(self, address: Address) -> Optional[int]:
                 balance = int(result[0]["amount"])
         return balance
 
+    def get_state(self, callable_name: str, *args) -> Optional[Any]:
+        """Call a specified function on the ledger API."""
+        response = self._try_get_state(callable_name, *args)
+        return response
+
+    @try_decorator(
+        "Encountered exception when trying get state: {}",
+        logger_method=_default_logger.warning,
+    )
+    def _try_get_state(self, callable_name: str, *args) -> Optional[Any]:
+        """Try to call a function on the ledger API."""
+        result = None  # type: Optional[Any]
+        query = "/".join(args)
```

Comment:
are we sure this is correct and its now a parameter list or similar?

Reply:
This is based on the cosmos REST API specification here: https://cosmos.network/rpc/v0.38.3, which takes a path as defined above. I took the convention of defining the root of the path (`txs`, `blocks`, etc.) as the `callable_name` and the rest of the path as arguments, partly to keep some consistency with the ethereum ledger interface, but it could also be handled as a simple list of arguments.

Reply:
It'd probably move that description into line 401 so we have it near the code for future devs.